app/Http/Controllers/ConnectionController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+use App\Connection;
+use Illuminate\Support\Facades\Auth;
+
+class ConnectionController extends Controller
+{
+    public function show($id)
+    {
+        $user = User::findOrFail($id);
+        $incoming = Connection::getIncomingConnections($user);
+        $outgoing = Connection::getOutgoingConnections($user);
+
+        return view('users.connections')
+            ->with('user', $user)
+            ->with('incoming', $incoming)
+            ->with('outgoing', $outgoing);
+    }
+
+    public function add($id)
+    {
+        $to_user = User::findOrFail($id);
+        $user = Auth::user();
+
+        if($user == null)
+        {
+            return(redirect()->route('login'));
```

Comment:
In the routes file add the `auth` middleware to this method and it'll handle the checking and redirecting.

Reply:
Also, cool.